From 17cf704b747b5b66c8aa6e798ea2f5f9f21f28c6 Mon Sep 17 00:00:00 2001
From: Shunsuke Kataoka <shunsuke.kataoka.df@renesas.com>
Date: Fri, 21 Jun 2013 14:41:11 +0900
Subject: [PATCH 0658/1083] ARM: shmobile: r8a7790: Add descriptor operating
 mode support for audma

This is example for the descriptor operating mode.
"desc_mode" is set by 2, scu_pcm driver use the Descriptor repeat mode.
"desc_offset" is set by 0x0 in playback, and 0x100 in capture.
"desc_stepnum" is set by 4.

Signed-off-by: Shunsuke Kataoka <shunsuke.kataoka.df@renesas.com>
(cherry picked from commit 143a09da9793329322d942117cd93ec0f2b7e4e6)

Signed-off-by: Ryo Kataoka <ryo.kataoka.wt@renesas.com>
---
 arch/arm/mach-shmobile/setup-r8a7790.c | 15 +++++++++++++++
 1 file changed, 15 insertions(+)

diff --git a/arch/arm/mach-shmobile/setup-r8a7790.c b/arch/arm/mach-shmobile/setup-r8a7790.c
index 1adca0a..904493a 100644
--- a/arch/arm/mach-shmobile/setup-r8a7790.c
+++ b/arch/arm/mach-shmobile/setup-r8a7790.c
@@ -936,26 +936,41 @@ static const struct sh_dmadesc_slave_config r8a7790_audma_slaves[] = {
 		.addr		= 0xec241008,
 		.chcr		= CHCR_TX(XMIT_SZ_32BIT),
 		.mid_rid	= 0x01,
+		.desc_mode	= 2,
+		.desc_offset	= 0x0,
+		.desc_stepnum	= 4,
 	}, {
 		.slave_id	= SHDMA_SLAVE_PCM_MEM_SRC0,
 		.addr		= 0xec000000,
 		.chcr		= CHCR_TX(XMIT_SZ_32BIT),
 		.mid_rid	= 0x85,
+		.desc_mode	= 2,
+		.desc_offset	= 0x0,
+		.desc_stepnum	= 4,
 	}, {
 		.slave_id	= SHDMA_SLAVE_PCM_SSI1_MEM,
 		.addr		= 0xec24104c,
 		.chcr		= CHCR_RX(XMIT_SZ_32BIT),
 		.mid_rid	= 0x04,
+		.desc_mode	= 2,
+		.desc_offset	= 0x100,
+		.desc_stepnum	= 4,
 	}, {
 		.slave_id	= SHDMA_SLAVE_PCM_SRC1_MEM,
 		.addr		= 0xec004400,
 		.chcr		= CHCR_RX(XMIT_SZ_32BIT),
 		.mid_rid	= 0x9c,
+		.desc_mode	= 2,
+		.desc_offset	= 0x100,
+		.desc_stepnum	= 4,
 	}, {
 		.slave_id	= SHDMA_SLAVE_PCM_CMD1_MEM,
 		.addr		= 0xec008400,
 		.chcr		= CHCR_RX(XMIT_SZ_32BIT),
 		.mid_rid	= 0xbe,
+		.desc_mode	= 2,
+		.desc_offset	= 0x100,
+		.desc_stepnum	= 4,
 	},
 };
 
-- 
1.8.3.2

